recovery.go

```diff
@@ -8,12 +8,17 @@ import (
 	"runtime"
 )
 
+type ErrorHandlerFunc interface {
+	HandleError(err interface{}, report bool)
+}
+
 // Recovery is a Negroni middleware that recovers from any panics and writes a 500 if there was one.
 type Recovery struct {
-	Logger     *log.Logger
-	PrintStack bool
-	StackAll   bool
-	StackSize  int
+	Logger           *log.Logger
+	PrintStack       bool
+	ErrorHandlerFunc ErrorHandlerFunc
```

Comment:
Can this be switched to the following, so that the surface area of the library doesn't have to expand?

``` go
        ErrorHandlerFunc func(interface{})
```